### PR TITLE
feat(styles): Update form-control and form-select sizes

### PR DESCRIPTION
**Pull request by imagoiq**
### What
- Update form-control, form-select sizes to match Figma
- Support small size with floating label
- Update floating label color of form-control to match Figma

Note: support for deprecated sizes is not consistent (for example form-control-rg with invalid state as a greater padding with input[type=date]). Date icon on Firefox are misaligned (they cannot be [restyled since v109](https://bugzilla.mozilla.org/show_bug.cgi?id=1812397))

### Review
Check all similar components: e.g. Textarea, File Input, …

### CODE REVIEW

**Review by oliverschuerch — COMMENTED** [review]
The text of the small select variant gets cropped (see letter `p`) and the vertical alignment looks wrong (at least in chrome). I think we need to reduce the line-height there or for all small variants.
image

**Review by oliverschuerch — CHANGES_REQUESTED** [review]
Both stories (select and input) should set the `size` control to the `large` value. As this will be the new default size.

**Review by oliverschuerch — CHANGES_REQUESTED** [review]
We should remove the deprecated sizes (`Regular` and `Middle`) from the size control in storybook and describe the deprecation of them in a alert before or after the controls.

They still need to work

This would also enable us to use the size without a class as the new default.

Mapping:
| Old size | New Size | CssClass |
| - | - | - |
| `Small` | `Small` | `form-control-sm` |
| `Regular` | deprecated | `form-control-rg` |
| `Middle` | `Large` | `none`
| `Large` | deprecated | `form-control-lg` |

What do you think?

**Comment by imagoiq** [comment]
> We should remove the deprecated sizes (`Regular` and `Middle`) from the size control in storybook and describe the deprecation of them in a alert before or after the controls.
> 
> They still need to work
> 
> This would also enable us to use the size without a class as the new default.
> 
> Mapping:
> Old size 	New Size 	CssClass
> `Small` 	`Small` 	`form-control-sm`
> `Regular` 	deprecated 	`form-control-rg`
> `Middle` 	`Large` 	`none`
> `Large` 	deprecated 	`form-control-lg`
> 
> What do you think?

Sure, but it will be handled with https://github.com/swisspost/design-system/issues/2225

File: packages/documentation/src/stories/components/forms/input/input.stories.ts
[the remark this answers is not in the document]
Comment (imagoiq) [reply packages/documentation/src/stories/components/forms/input/input.stories.ts:1]: I agree, but they will automatically disappear when we'll remove the deprecated size, so it is easier to wait then to workaround for the meantime.

[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/documentation/src/stories/components/forms/input/input.stories.ts:1]: We will only remove this deprecated sizes when we release v7, which could be at the end of this year or even later. So in the meantime I prefer to not confuse our users with non existing input sizes.

**Comment by oliverschuerch** [comment]
> > We should remove the deprecated sizes (`Regular` and `Middle`) from the size control in storybook and describe the deprecation of them in a alert before or after the controls.
> > They still need to work
> > This would also enable us to use the size without a class as the new default.
> > Mapping:
> > Old size 	New Size 	CssClass
> > `Small` 	`Small` 	`form-control-sm`
> > `Regular` 	deprecated 	`form-control-rg`
> > `Middle` 	`Large` 	`none`
> > `Large` 	deprecated 	`form-control-lg`
> > What do you think?
> 
> Sure, but it will be handled with #2225

Sure, but this is a breaking change who will be implemented in the far future.

File: packages/documentation/src/stories/components/forms/input/input.stories.ts
[the remark this answers is not in the document]
Comment (imagoiq) [reply packages/documentation/src/stories/components/forms/input/input.stories.ts:1]: Sorry I thought that you were speaking about the snapshots, but you are speaking about the controls. I don't see how to do it using if condition in the argtypes especially as you cannot have multiple if (https://github.com/storybookjs/storybook/issues/21281). Do you have any idea?

**Review by oliverschuerch — APPROVED** [review]

**Pull request merged**